### PR TITLE
Driveable Death Explosion Changes

### PR DESCRIPTION
**Pull request by conman180**
I copied some code from ShootableType.java and EntityGrenade.java over to DriveableType and EntiyDriveable to give control over the explosion if IsExplosionWhenDestroyed is set to true.

This would allow crashing a vehicle to be destructive rather than just making blocks disappear when hit.
An example that could be made is Kamikaze aircraft or a car bomb.

Hopefully this can make a fun addition to the mod.

### CODE REVIEW

**Comment by conman180** [comment]
Here is an example clip I made to show it works However plane physics are different here to to my personal meddling with the flight controller. TBH I had  a bit too much fun while testing this.
PlaneShootdownWithBoom

**Review by GoldSloth — COMMENTED** [review]
Nice work, these provide a lot of customisability to the explosion. One thing I'd request changed though, is to prefix both the variable names and the config keywords to indicate that they're for configuring the death explosion, otherwise it's a bit vague imo. It'd also be great if you could add these to the config reference too, but I'm happy to do that otherwise.

**Comment by GoldSloth** [comment]
I'll also take a look at the explosion logic later, I'm not sure both types of explosions are needed - but I'm sure it works regardless. Many thanks for your contributions!

**Comment by conman180** [comment]
I altered those commits, should be a separate set of config options for death explosions. I also added them to the config reference

File: src/main/java/com/flansmod/common/driveables/EntityDriveable.java [hunk 1]
```diff
@@ -2471,7 +2472,36 @@ public void checkParts() {
...
+//Create a flans mod explosion rather than a standard MC one. allows control over death boom   	
+    	        {
+        	        new FlansModExplosion(worldObj, this, null, type, posX, posY, posZ,
+        		        	type.deathExplosionRadius, type.deathExplosionPower,TeamsManager.explosions && type.deathExplosionBreaksBlocks,
+        		        	type.deathExplosionDamageVsLiving, type.deathExplosionDamageVsPlayer, type.deathExplosionDamageVsPlane, type.deathExplosionDamageVsVehicle, seatNum, seatNum);
+        	
+        	        }
+        	        else
+        	        {
+        	        	worldObj.createExplosion(this, posX, posY, posZ, type.explosionRadius, TeamsManager.explosions && type.explosionBreaksBlocks);
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/driveables/EntityDriveable.java]: Sorry, just got a chance to look at this again. If I understand it correctly the new logic is:

```Java
if (type.IsExplosionWhenDestroyed) {
    // Flans big boom fancy explosion
} else {
    // Vanilla explosion
}
```
To keep consistent with the old way it worked, I'd  suggest removing the else, it's either a custom Flans explosion (People can remove the destructiveness if they need) or no explosion at all. Do you agree with that?

**Review by GoldSloth — CHANGES_REQUESTED** [review]
rip, forgot to submit this review last night .-.

File: src/main/java/com/flansmod/common/driveables/EntityDriveable.java (hunk 1 above)
Comment (conman180) [reply src/main/java/com/flansmod/common/driveables/EntityDriveable.java]: I agree. Ill remove the else

**Review by GoldSloth — APPROVED** [review]

**Comment by GoldSloth** [comment]
Awesome! Thank you for your work on this feature, I'll pull them both now.

**Pull request merged**